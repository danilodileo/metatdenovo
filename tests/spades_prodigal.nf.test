nextflow_pipeline {

    name "Test pipeline"
    script "../main.nf"
    tag "pipeline"
    profile "test_spades_prodigal"

    test("-profile test_spades_prodigal") {

        when {
            params {
                outdir = "$outputDir"
            }
        }

        then {
            // stable_name: All files + folders in ${params.outdir}/ with a stable name
            def stable_name = getAllFilesFromDir(params.outdir, relative: true, includeDir: true, ignore: ['pipeline_info/*.{html,json,txt}'])
            // stable_path: All files in ${params.outdir}/ with stable content
            def stable_path = getAllFilesFromDir(params.outdir, ignoreFile: 'tests/.nftignore')
            assertAll(
                { assert workflow.success },
                { assert snapshot(
                    // Number of successful tasks
                    workflow.trace.succeeded().size(),
                    // pipeline versions.yml file for multiqc from which Nextflow version is removed because we test pipelines on multiple Nextflow versions
                    removeNextflowVersion("$outputDir/pipeline_info/nf_core_metatdenovo_software_mqc_versions.yml"),
                    // All stable path name, with a relative path
                    stable_name,
                    // All files with stable contents
                    stable_path,
                    // Important files with unstable content
                    path("${params.outdir}/samtools/spades.SAMPLE1_PE.flagstat").readLines().size(),
                    path("${params.outdir}/summary_tables/spades.prodigal.counts.tsv.gz").linesGzip.findAll { l -> l.contains('orf') },
                    path("${params.outdir}/summary_tables/spades.prodigal.overall_stats.tsv.gz").linesGzip.size(),
                    path("${params.outdir}/summary_tables/spades.prodigal.overall_stats.tsv.gz").linesGzip.findAll { l -> l.contains('n_feature_count') },
                ).match() },
                { assert path("${params.outdir}/spades/spades.spades.log").readLines().any { l -> l.contains('======= SPAdes pipeline finished.') } },
                { assert path("${params.outdir}/spades/spades.assembly.gfa.gz").linesGzip.size() > 10000 },
                { assert path("${params.outdir}/spades/spades.transcripts.fa.gz").linesGzip.size() > 3500 },
                { assert path("${params.outdir}/samtools/spades.SAMPLE1_PE.idxstats").readLines().size() > 1700 },
                { assert path("${params.outdir}/prodigal/spades.prodigal.gff.gz").linesGzip.findAll { l -> l.contains('CDS') }.size() > 3000 },
                { assert path("${params.outdir}/summary_tables/spades.prodigal.counts.tsv.gz").linesGzip.size() > 3000 },
            )
        }
    }
}
