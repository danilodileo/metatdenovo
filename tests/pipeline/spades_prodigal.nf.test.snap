{
    "SAMPLE1_PE_1_val_1_fastqc.html": {
        "content": [
            "SAMPLE1_PE_1_val_1_fastqc.html:md5,89c71d5bf2499665b8c036a2ffae3bd7"
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-02-19T18:06:50.891439"
    },
    "spades.spades.log": {
        "content": [
            "spades.spades.log:md5,2a600b77ced65d94c8a6273a11514ec6"
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-02-19T18:06:50.903446"
    },
    "SAMPLE1_PE.bbmap.log": {
        "content": [
            "SAMPLE1_PE.bbmap.log:md5,9528fbb9ddf29b0b8952c1584fad19f2"
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-02-19T18:06:50.91241"
    },
    "multiqc_report.html": {
        "content": [
            "multiqc_report.html:md5,e81df4815b69ad17a19a888daca7bb7a"
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-02-19T18:06:50.948837"
    },
    "SAMPLE1_PE.featureCounts.tsv.summary": {
        "content": [
            "SAMPLE1_PE.featureCounts.tsv.summary:md5,3b293e6de265163459a8ef02cb86887a"
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-02-05T18:06:30.542728"
    }
}